### PR TITLE
Fix for covering channels query

### LINKED ISSUE #3822: Rewrite and guard against non-covering queries
The channels and all_docs queries are currently non-covering.  For channels this is due to an error in the query statement.  all_docs is known to be non-covering, but should be reviewed to identify whether a covering version is possible, due to https://issues.couchbase.com/browse/MB-30467. 

Along with fixing the query, we should put in place unit tests that retrieve the EXPLAIN for the query and validate that it's covering (to guard against future regressions).

### PR DESCRIPTION
**Pull request by adamcfraser**
Adds EXPLAIN support so that unit test can validate queries are covering.

Fixes #3822

### CODE REVIEW

File: db/query.go [hunk 1]
```diff
@@ -210,6 +210,20 @@ func (context *DatabaseContext) N1QLQueryWithStats(queryName string, statement s
 	return results, err
 }
 
+// N1QlExplain issues an EXPLAIN query for the given statement
+func (context *DatabaseContext) N1QLExplain(statement string, params interface{}, consistency gocb.ConsistencyMode) (results gocb.QueryResults, err error) {
```
Comment (bbrks) [inline db/query.go]: It doesn't look like this is used anywhere. Was this added for completion, or an old bit of code?

**Comment by adamcfraser** [comment]
Kicked off integration tests as sanity check on the query definition change:

http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/851/
http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/852/

File: db/query.go (hunk 1 above)
Comment (adamcfraser) [reply db/query.go]: It made more sense to push this down to bucket, and I forgot to clean this up.  Pushed up a commit to remove.

**Review by bbrks — APPROVED** [review]
Approving now, all good to merge after the last test finishes.

**Pull request merged**